[DealabsBridge] First version of the generic "Pepper" Bridge

sysadminstory opened the pull request:
- PepperBridgeAbstract is the class that contains all the code
- Any class extending PepperBridgeAbstract must only define the classic
  bridge constant, and the $lang variable which contains the specific
  website content
- Under the PepperBridgeAbstract class there is an example class with
  comments to implement a new Bridge
- DealabsBridge extends now PepperBridgeAbstract
- HotUKDealsBridge is the first new use of PepperBridgeAbstract

sysadminstory committed:
[DealabsBridge] First version of the generic "Pepper" Bridge

- PepperBridgeAbstract is the class that contains all the code
- Any class extending PepperBridgeAbstract must only define the classic
  bridge constant, and the $lang variable which contains the specific
  website content
- Under the PepperBridgeAbstract class there is an example class with
  comments to implement a new Bridge
- DealabsBridge extends now PepperBridgeAbstract
- HotUKDealsBridge is the first new use of PepperBridgeAbstract

sysadminstory committed:
[DealabsBridge] Coding policy fix and new Pepper Bridge

- MydealsBridge is the second Pepper Bridge
- Added some function to handle relative date while sorting by hottest
  deals
- There could be multiple way a relative date could be recognized :
  switched to an array of strings

sysadminstory committed:
[DealabsBridge] Coding policy fix

- Yeah, again, coding policy fix

teromene commented on bridges/DealabsBridge.php:
```diff
@@ -458,19 +551,189 @@ private function relativeDateToTimestamp($str) {
-				break;
-			default: // Return default value
-				return self::NAME;
+		case $this->i8n('context-keyword'):
```
Indentation withing the switch

teromene commented on bridges/DealabsBridge.php:
```diff
@@ -458,19 +551,189 @@ private function relativeDateToTimestamp($str) {
+
+/*
+
+class ExampleDealsBridge extends PepperBridgeAbstract {
```
The example should probably not be in the file. You should create a wiki page for this.

teromene reviewed (commented):
Thanks for the PR ! Just a few remarks before we can get this merged !

sysadminstory committed:
[DealabsBridge] Coding policy fixes & doc migration

- Changed the switch indentation as asked
- Deletion of the the example class : doc will be in the Project Wiki

sysadminstory commented:
I followed your advice :
* I removed the example class, and created a draft of Wiki page here : https://github.com/sysadminstory/rss-bridge/wiki (If it suits you, maybe you can tell me which page I should create ;) )
* I modified the switch indentation

teromene commented on bridges/DealabsBridge.php:
```diff
@@ -263,15 +344,15 @@ private function getPrix($deal)
 	 */
 	private function getLivraison($deal)
```
As your bridge now supports international websites, I would rename this to ```getDelivery```. Same for the rest.

sysadminstory replied:
Good idea ! I did not read the function name anymore, only the content :)

teromene commented on bridges/DealabsBridge.php:
```diff
@@ -458,18 +551,38 @@ private function relativeDateToTimestamp($str) {
-				$groupe = array_search($this->getInput('groupe'), $values);
-				return self::NAME . ' - Groupe : '. $groupe;
+			case $this->i8n('context-group'):
+				$values = static::PARAMETERS[$this->i8n('context-group')]['group']['values'];
```
Why not using getParameter here ?

sysadminstory replied:
Because I didn't know it existed ! As I know it now, I fixed it !

teromene reviewed (changes requested):
Just a last thing before we can merge.

sysadminstory committed:
[DealabsBridge] Renamed function to english

All function name use englis to be more international

sysadminstory committed:
[DealabsBridge] Use the BridgeAbstract functions

Change to the BridgeAbstract function getParameters instead of doing the
same as this function

teromene commented:
Thansk 👍
